src/commands/clone.js

```diff
@@ -41,10 +42,37 @@ function clone (opts) {
   var onExit = require('../lib/exit')
   var parseArgs = require('../parse-args')
   var debug = require('debug')('dat')
-
   var parsed = parseArgs(opts)
+
   opts.key = parsed.key || opts._[0] // pass other links to resolver
   opts.dir = parsed.dir
+
+  // Resolves dat.json path and parses dat url from it.
+  if (fs.existsSync(opts.key)) {
+    var datPath
+    var neatPath = neatLog(archiveUI, { logspeed: opts.logspeed, quiet: opts.quiet })
+    neatPath.use(trackArchive)
+    neatPath.use(discoveryExit)
+    neatPath.use(onExit)
+    if (path.basename(opts.key) === 'dat.json') { // handles paths for 'dat clone' & 'dat clone .'
+      datPath = path.resolve(opts.key)
+    } else if (fs.lstatSync(opts.key).isDirectory()) {
+      datPath = path.resolve(opts.key + '/dat.json')
+    } else {
+      neatPath.use(function (state, bus) {
+        return bus.emit('exit:error', 'Could not resolve link from dat.json on path: ' + path.resolve(opts.key))
+      })
+    }
+    if (!fs.existsSync(datPath)) {
```

Comment:
Ah I see, its checking if the `dat.json` exists, not just the path. Same [comment as above](https://github.com/datproject/dat/pull/845/files#r132571106), we should assume this is a key and fail later.